cli/diff.rs

```diff
@@ -27,146 +165,60 @@ fn fmt_rem_text_highlight(x: &str) -> String {
   colors::white_on_red(x).to_string()
 }
 
-fn write_line_diff(
-  diff: &mut String,
-  orig_line: &mut usize,
-  edit_line: &mut usize,
-  line_number_width: usize,
-  orig: &mut String,
-  edit: &mut String,
-) {
-  let split = orig.split('\n').enumerate();
-  for (i, s) in split {
-    diff.push_str(&format!(
-      "{:width$}{} ",
-      *orig_line + i,
-      colors::gray(" |"),
-      width = line_number_width
-    ));
-    diff.push_str(&fmt_rem());
-    diff.push_str(s);
-    diff.push('\n');
-  }
-
-  let split = edit.split('\n').enumerate();
-  for (i, s) in split {
-    diff.push_str(&format!(
-      "{:width$}{} ",
-      *edit_line + i,
-      colors::gray(" |"),
-      width = line_number_width
-    ));
-    diff.push_str(&fmt_add());
-    diff.push_str(s);
-    diff.push('\n');
-  }
-
-  *orig_line += orig.split('\n').count();
-  *edit_line += edit.split('\n').count();
-
-  orig.clear();
-  edit.clear();
-}
-
-/// Print diff of the same file_path, before and after formatting.
-///
-/// Diff format is loosely based on Github diff formatting.
-pub fn diff(orig_text: &str, edit_text: &str) -> String {
-  let lines = edit_text.split('\n').count();
-  let line_number_width = lines.to_string().chars().count();
```

Comment:
Bug here fixed in new code. Should take into account the max of orig_text and edit_text line count.